Lib/posixpath.py

```diff
@@ -443,6 +443,17 @@ def _realpath(filename, strict=False, sep=sep, curdir=curdir, pardir=pardir,
             newpath = path + name
         else:
             newpath = path + sep + name
+        if newpath in seen:
```

Comment:
Is this faster for `strict` mode?

```suggestion
        if maxlinks is not None and newpath in seen:
```